File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<int>>mos;
        int n = strs.size();
        vector<string>temp(strs.begin(),strs.end());
        for (int i=0;i<n;i++){
            sort(temp[i].begin(),temp[i].end());
            mos[temp[i]].push_back(i);
        }
        vector<vector<string>>vos;
        for (auto it : mos){
            vector<string>t;
            for (auto ls : it.second){
                t.push_back(strs[ls]);
            }
            vos.push_back(t);
        }
        return vos;
    }
};

File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let mp = new Map();
    for (let word of strs){
        let sorted = word.split('').sort().join('');
        if (!mp.has(sorted)){
            mp.set(sorted,[])
        }
        mp.get(sorted).push(word);
    }
    return Array.from(mp.values());
};